pyVoIP/SIP.py

```diff
@@ -583,16 +583,24 @@ def start(self):
      
   def stop(self):
     self.NSD = False
-    self.registerThread.cancel()
-    self.deregister()
-    self.s.close()
-    self.out.close()
+    if self.registerThread:
+      self.registerThread.cancel()
+      self.deregister()
+    self._close_sockets()
+
+  def _close_sockets(self):
+    if hasattr(self, 's'):
```

Comment:
if statements could be a on-liner
`if hasattr(self, 's') and self.s:`
this will not throw an exception in any way if `self.s` does not exist.
separating `if self.s` only makes sense if you want to react on some value of `self.s`. Here it's "only" close socket if it's present